### DISCUSSION TITLE
How to convert bootloader elf file into bin file

### DISCUSSION THREAD

**Discussion by alf45tar** [espressif/arduino-esp32#8140] [Q&A] [answered]
In the latest version of the framework bootloader bin files are not anymore provided by the Arduino framework but only elf files are present.
How to convert elf files into bin files?

Thanks
alf45tar

**Comment by alf45tar** [comment]
https://docs.espressif.com/projects/esptool/en/latest/esp32/esptool/basic-commands.html#convert-elf-to-binary-elf2image

**Comment by alf45tar** [comment]
I am using PlatformIO on macOS and the following command:

`~/.platformio/penv/bin/python ~/.platformio/packages/tool-esptoolpy/esptool.py --chip ESP32 elf2image ~/.platformio/packages/framework-arduinoespressif32/tools/sdk/esp32/bin/bootloader_dio_40m.elf`

**Comment by mrengineer7777** [comment, marked as answer]
@alf45tar Yes, that is the correct way to convert .elf to .bin files.  I'm not familiar with Arduino core providing BIN files even in earlier versions.  I get my binaries (including bootloader) from here:
C:\Users\USERNAME\Documents\PlatformIO\Projects\PROJECT_NAME\.pio\build\WORKSPACE_NAME

**Reply by mrengineer7777** [reply]
All files if you are making your own flash script:
python		Locally installed python version.  See also C:\Users\USERNAME\.platformio\penv\Scripts\python.exe.
esptool		C:\Users\USERNAME\.platformio\packages\tool-esptoolpy\esptool.py
boot_app0		C:\Users\USERNAME\.platformio\packages\framework-arduinoespressif32\tools\partitions\boot_app0.bin
bootloader		C:\Users\USERNAME\Documents\PlatformIO\Projects\PROJECTNAME\.pio\build\PROJECTNAME-Release\bootloader.bin
partitions		C:\Users\USERNAME\Documents\PlatformIO\Projects\PROJECTNAME\.pio\build\PROJECTNAME-Release\partitions.bin
firmware		C:\Users\USERNAME\Documents\PlatformIO\Projects\PROJECTNAME\.pio\build\PROJECTNAME-Release\firmware.bin
spiffs		C:\Users\USERNAME\Documents\PlatformIO\Projects\PROJECTNAME\.pio\build\PROJECTNAME-Release\spiffs.bin

**Reply by Sydfg** [reply]
192.168.1.128

**Reply by Sydfg** [reply]
192.168.1.128l
